src/compiler/scala/tools/nsc/javac/JavaParsers.scala

```diff
@@ -1092,6 +1092,14 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
     /** CompilationUnit ::= [[Annotation] package QualId semi] {Import} {TypeDecl} //TopStatSeq
      */
     def compilationUnit(): Tree = {
+      var compact = false
+      def typeDeclOrCompact(mods: Modifiers): List[Tree] = in.token match {
+        case ENUM | INTERFACE | AT | CLASS | RECORD => typeDecl(mods)
+        case _ =>
+          val ts = termDecl(mods, CLASS)
+          if (ts.nonEmpty) compact = true
+          Nil
+      }
```

Comment:
I might have generalized `typeDeclOrCompactTerm(mods)(orElse: Tree)` which `typeDecl` would call.

I might have orElse throw here to long jump back to a  try in compilationUnit.

That would be to avoid listing tokens at L1097. OTOH, the method is so small this is fine.

There is a `return` in that `interactive` code, so maybe I have early return on the brain.